keyboards/takashicompany/baumkuchen/keymaps/via/keymap.c

```diff
@@ -0,0 +1,68 @@
+// Copyright 2023 takashicompany (@takashicompany)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
+        LT(1, KC_ESC), KC_Q, KC_W, KC_E, 
+        KC_TAB, KC_A, KC_S, KC_D,
+        KC_LSFT, KC_Z, KC_X, KC_C, KC_V,
+        KC_WH_U, KC_WH_D
+    ),
+
+    [1] = LAYOUT(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, MO(0), KC_TRNS,
+        KC_TRNS,KC_TRNS
+    ),
+
+    [2] = LAYOUT(
+        RGB_TOG, RGB_MOD, KC_TRNS, QK_BOOT,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_LSFT, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS,KC_TRNS
+    )
+};
+
+//------------------------------------------------------------------------------
+// Rotary Encoder
+//------------------------------------------------------------------------------
+static uint8_t  encoder_state[NUM_ENCODERS] = {0};
+static keypos_t encoder_cw[NUM_ENCODERS]    = ENCODERS_CW_KEY;
+static keypos_t encoder_ccw[NUM_ENCODERS]   = ENCODERS_CCW_KEY;
+
+void encoder_action_unregister(void) {
+    for (int index = 0; index < NUM_ENCODERS; ++index) {
+        if (encoder_state[index]) {
+            keyevent_t encoder_event = (keyevent_t) {
+                .key = encoder_state[index] >> 1 ? encoder_cw[index] : encoder_ccw[index],
+                .pressed = false,
+                .time = (timer_read() | 1)
+            };
+            encoder_state[index] = 0;
+            action_exec(encoder_event);
+        }
+    }
+}
+
+void encoder_action_register(uint8_t index, bool clockwise) {
+    keyevent_t encoder_event = (keyevent_t) {
+        .key = clockwise ? encoder_cw[index] : encoder_ccw[index],
+        .pressed = true,
+        .time = (timer_read() | 1)
+    };
+    encoder_state[index] = (clockwise ^ 1) | (clockwise << 1);
+    action_exec(encoder_event);
+}
+
+void matrix_scan_kb(void) {
+    encoder_action_unregister();
+    matrix_scan_user();
+}
```

Comment:
This is not allowed in the repo now, and <https://docs.qmk.fm/#/feature_encoders?id=encoder-map> used instead.